src/commands/appSettings/downloadAppSettings.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebSiteManagementModels } from "azure-arm-website";
+import { DotenvParseOutput } from "dotenv";
+import * as fse from 'fs-extra';
+import * as os from 'os';
+import * as vscode from 'vscode';
+import { AppSettingsTreeItem, SiteClient } from "vscode-azureappservice";
+import { envFileName } from "../../constants";
+import { ext } from "../../extensionVariables";
+import * as workspaceUtil from '../../utils/workspace';
+import { confirmOverwriteSettings } from "./confirmOverwriteSettings";
+import { getLocalEnvironmentVariables } from './getLocalEnvironmentVariables';
+
+export async function downloadAppSettings(node?: AppSettingsTreeItem): Promise<void> {
+    if (!node) {
+        node = <AppSettingsTreeItem>await ext.tree.showTreeItemPicker(AppSettingsTreeItem.contextValue);
+    }
+
+    const client: SiteClient = node.root.client;
+
+    const message: string = 'Select the destination file for your downloaded settings.';
+    const envVarPath: string = await workspaceUtil.selectWorkspaceFile(ext.ui, message, () => envFileName);
+    const envVarUri: vscode.Uri = vscode.Uri.file(envVarPath);
+
+    await node.runWithTemporaryDescription('Downloading...', async () => {
+        ext.outputChannel.appendLine(`Downloading settings from "${client.fullName}"...`);
```

Comment:
Do we really need to keep duplicating these messages in both the notifications and output channel? Seems like it just makes the code harder to read without providing much value

Reply:
Do you mean we should make a constant for `Downloading settings from "${client.fullName"`?  I'm not clear what you're trying to suggest here

Reply:
I'm saying get rid of this line:
```
ext.outputChannel.appendLine(`Downloading settings from "${client.fullName}"...`);
```

It's a somewhat common pattern that you do, but it's starting to feel redundant and can honestly make the code look a little confusing.